### PR TITLE
Refactor interactors

### PR DESCRIPTION
**Pull request by kgolezardi**

### CODE REVIEW

File: review/accounts/interactors.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+def user_login(requset, username, password):
+    user = authenticate(username=username, password=password)
+    if user is not None:
+        if user.is_active:
+            login(requset, user)
+            return user
+    return None
+
+
+def user_logout(request):
```
Comment (smmoosavi) [inline review/accounts/interactors.py]: move back

**Comment by kgolezardi** [comment]
https://docs.djangoproject.com/en/3.0/topics/db/queries/#complex-lookups-with-q-objects

File: review/core/interactors/person_review.py [hunk 2]
```diff
@@ -1,7 +1,13 @@
+from django.db.models import Q
+
 from core.models import PersonReview, ProjectReview
 
 
-def can_review(reviewer, reviewee):
+def can_review_person(reviewer, reviewee):
```
Comment (smmoosavi) [inline review/core/interactors/person_review.py]: rename reviewer to user

File: review/core/interactors/person_review.py [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
+    # TODO: need improvement based on current phase
+    if not reviewer.is_authenticated:
+        return True
```
Comment (smmoosavi) [inline review/core/interactors/person_review.py]: return false?

File: review/core/interactors/person_review.py [hunk 4]
```diff
@@ -35,32 +38,19 @@ def save_person_review(reviewee, reviewer, **kwargs):
 
 
 def get_all_person_reviews(user):
-    # TODO: need improvement based on current phase
-    return PersonReview.objects.filter(reviewer=user)
+    return PersonReview.objects.filter(Q(reviewer=user) | Q(reviewer=user))
```
[the remark this answers is not in the document]
Comment (smmoosavi) [reply review/core/interactors/person_review.py]: only `reviewer=user`

File: review/core/schema/save_project_review_mutation.py [hunk 5]
```diff
@@ -15,23 +15,20 @@ class Input:
...
 
     project_review = graphene.Field(ProjectReviewNode)
 
-    # TODO: Login requied
     @classmethod
     def mutate_and_get_payload(cls, root, info, **args):
         project = get_node(args['project_id'], info, Project)
         reviewee = info.context.user
 
         reviewers = [get_node(reviewer_id, info, User) for reviewer_id in args['reviewers_id']]
```
Comment (smmoosavi) [inline review/core/schema/save_project_review_mutation.py]: use args.get('reviewers_id')

support null reviewers_id

**Pull request merged**